src/cmd/singularity/cli/actions.go

```diff
@@ -326,15 +326,23 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 	}
 
 	if Nvidia {
-		NvidiaBindPaths, err := nvidiautils.GetNvidiaBindPath(buildcfg.SINGULARITY_CONFDIR)
+		libs, bins, err := nvidiautils.GetNvidiaPath(buildcfg.SINGULARITY_CONFDIR)
 		if err != nil {
 			sylog.Infof("Unable to capture nvidia bind points: %v", err)
 		} else {
-			if len(NvidiaBindPaths) == 0 {
+			if len(bins) == 0 {
+				sylog.Infof("Could not find any NVIDIA binaries on this host!")
+			} else {
+				if IsWritable {
+					sylog.Warningf("NVIDIA binaries may not be bound with --writable")
+				}
+				BindPaths = append(BindPaths, bins...)
+			}
+			if len(libs) == 0 {
 				sylog.Warningf("Could not find any NVIDIA libraries on this host!")
 				sylog.Warningf("You may need to edit %v/nvliblist.conf", buildcfg.SINGULARITY_CONFDIR)
 			} else {
-				BindPaths = append(BindPaths, NvidiaBindPaths...)
+				engineConfig.SetLibrariesPath(libs)
```

Comment:
We have users that rely on the `SINGULARITY_CONTAINLIBS` environment variable to add other libraries at runtime as well.  Afaik, this includes things to get MPI running properly.  Those users could also use `SINGULARITY_BINDPATH` but of course that option will break if the container is executed with the `--writable` option.  So imho we should also allow users to specify the `SINGULARITY_CONTAINLIBS` and append its contents to `libs` here.  

@bauerm97 is concerned about doing this because he is worried this will allow a user to bind files even if an admin has specified no user bind control in the conf file.  But afaik, if the conf file is appropriate checked in `container.go` that should be a non-issue. 

However, looking at the new code in `container.go`, it seems like the conf file is being checked but is not returning if user bind is turned off.  A warning is being issued and then the function is proceeding as normal.  Am I missing something?

Reply:
I will include discussed fix to handle SINGULARITY_CONTAINLIBS variable